app/models/user.rb

```diff
@@ -225,6 +225,12 @@ def unique_role_names
     roles.map(&:name).uniq
   end
 
+  memoize def reporting_policy_for_project(project_id:)
```

Comment:
Can you add a comment describing what the goal of this method is?

Reply:
I don't think we should memoize this as policy_for is already memoized.

```suggestion
 def reporting_policy_for_project(project_id:)
```

Reply:
Also `GrdaWarehouse::Hud::Client#project_pii_provider` is doing something kind of similar. Maybe we could centralize it? That let's us respect the include_pii_in_detail_downloads configuration too


```diff
diff --git a/app/models/grda_warehouse/hud/client.rb b/app/models/grda_warehouse/hud/client.rb
index a198809f41..9466d9f88a 100644
--- a/app/models/grda_warehouse/hud/client.rb
+++ b/app/models/grda_warehouse/hud/client.rb
@@ -1327,18 +1327,7 @@ module GrdaWarehouse::Hud
 
     # pii provider for use in reports and bulk view
     def project_pii_provider(project:, user:, mode:)
-      allowed = false
-      case mode.to_sym
-      when :download
-        allowed = ::GrdaWarehouse::Config.get(:include_pii_in_detail_downloads)
-      when :browse
-        allowed = true
-      else
-        raise ArgumentError, "Bad mode #{mode}"
-      end
-
-      policy = user.policy_for(project, policy_class: GrdaWarehouse::AuthPolicies::ProjectPiiPolicy) if allowed
-      policy ||= GrdaWarehouse::AuthPolicies::DenyPiiPolicy.instance
+      policy = user.reporting_policy_for_project(project, mode: mode)
       GrdaWarehouse::PiiProvider.new(self, policy: policy)
     end
 
diff --git a/app/models/user.rb b/app/models/user.rb
index e07b485c6d..65a616cb32 100644
--- a/app/models/user.rb
+++ b/app/models/user.rb
@@ -225,10 +225,21 @@ class User < ApplicationRecord
     roles.map(&:name).uniq
   end
 
-  memoize def reporting_policy_for_project(project_id:)
-    return GrdaWarehouse::AuthPolicies::LegacyPiiPolicy.instance if project_id.blank?
+  def reporting_policy_for_project(project_id:, mode: :browse)
+    return GrdaWarehouse::AuthPolicies::LegacyPiiPolicy.instance if project_id.nil?
+
+    allowed = false
+    case mode.to_sym
+    when :download
+      allowed = ::GrdaWarehouse::Config.get(:include_pii_in_detail_downloads)
+    when :browse
+      allowed = true
+    else
+      raise ArgumentError, "Bad mode #{mode}"
+    end
 
-    policy_for(project_id.to_i, policy_class: GrdaWarehouse::AuthPolicies::ProjectPiiPolicy)
+    policy = policy_for(project, policy_class: GrdaWarehouse::AuthPolicies::ProjectPiiPolicy) if allowed
+    policy || GrdaWarehouse::AuthPolicies::DenyPiiPolicy.instance
   end
 
   memoize def policy_for(resource, policy_class: nil)
```